src/SoftRobots/component/constraint/model/CableModel.cpp

```diff
@@ -41,12 +40,158 @@ namespace component
 namespace constraintset
 {
 
-using namespace sofa::defaulttype;
+
+using sofa::defaulttype::Vec2Types;
+using sofa::type::Vector2;
+
+template<>
+unsigned int CableModel<Vec2Types>::computeClosestIndice(Coord position)
+{
+    double closest_distance = (position - Coord(m_topology->getPX(0), m_topology->getPY(0))).norm();
+    unsigned int closest_vertice = 0;
+    for(unsigned int j=1; j<m_topology->getNbPoints(); j++)
+    {
+        double distance = (position - Coord(m_topology->getPX(j), m_topology->getPY(j))).norm();
+        if(distance < closest_distance) 
+        {
+            closest_distance = distance;
+            closest_vertice = j;
+        }
+    } 
+   return closest_vertice;
+}
+
+
+template<>
+void CableModel<Vec2Types>::computePointsActionArea()
+{
+    // Implementing continuous Dijkstra as a geodesic distance measure
+    ReadAccessor<Data<VecCoord>> cablePositions = m_state->readPositions();
+    unsigned int nbCenters = d_indices.getValue().size();
+    const SetIndexArray &indices = d_indices.getValue();
+    ReadAccessor<Data<vector<Real>>> maxDistance = d_radii;
+    vector<Real> m_totalRatio;
+    m_areaIndices.clear();
+    m_areaIndices.resize(nbCenters);
+    m_ratios.clear();
+    m_ratios.resize(nbCenters);
+    m_totalRatio.clear();
+    m_totalRatio.resize(nbCenters);
+
+    // Find closest indice on surface topology
+    SetIndexArray m_closestCenterIndices;
+    if (nbCenters != 0)
+    {
+        m_closestCenterIndices.resize(nbCenters);
+        for(unsigned int i=0; i<nbCenters; i++)
+            m_closestCenterIndices[i] = computeClosestIndice(cablePositions[indices[i]]);
```

Comment:
This is inefficient.

The most efficient way would be the following:
```
m_closestCenterIndices.reserve(nbCenters);
for(const auto id : indices)
{
    m_closestCenterIndices.push_back(computeClosestIndice(cablePositions[id]));
}
```

Note that you don't have to check that `nbCenters` is 0 or not.